test/e2e/specs/all-features-confirmation/record/presentation.spec.js

```diff
@@ -114,16 +114,20 @@ describe('View existing record,', function() {
 
             browser.wait(function() {
                 return chaisePage.recordPage.getRelatedTablesWithPanel().count().then(function(ct) {
+                    console.log("ct: "+ct);
                     return (ct=testParams.no_related_data.tables_order.length);
                 });
             }, browser.params.defaultTimeout);
             var showAllRTButton = chaisePage.recordPage.getShowAllRelatedEntitiesButton();
 
             chaisePage.recordPage.getRelatedTablesWithPanel().count().then(function(count) {
+                browser.sleep(2000);
+                console.log(count);
                 expect(count).toBe(testParams.no_related_data.tables_order.length, "Number of related tables is not correct");
 
                 return chaisePage.recordPage.getRelatedTableTitles();
             }).then(function(headings) {
+                browser.sleep(2000);
```

Comment:
A sleep statement here won't make a difference. The `headings` have already been fetched. This sleep is just delaying the check against the already available `headings` value.

If the problem is that headings aren't always correct, maybe we need to wait for all the related tables to be loaded like I stated above.